### PR TITLE
feat(updater): suggested updates come from syncing merkle tree to avoid updater state reads

### LINKED ISSUE #102: fix(middlewares): stop relying on updater timelagged state reads
- rollback ethers version to remove last updater patch (adds timelag for contract state reads)
- produce updates from local tree of messages which uses indexing timelag

### PR DESCRIPTION
**Pull request by ltchang2019**
- addressing `timelag x nonce manager` errors by removing state reads from updater
- adds syncing incremental merkle and has updater use it for producing updates
- **this will evantually become part of the home model when we do the actor-model refactor**

Closes #102

### CODE REVIEW

File: nomad-base/src/merkle_sync.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+};
+use std::{sync::Arc, time::Duration};
+use tokio::{sync::RwLock, task::JoinHandle, time::sleep};
+use tracing::{debug, error, info, info_span, instrument::Instrumented, Instrument};
+
+use crate::NomadDB;
+
+/// Self-syncing light merkle tree. Polls for new messages and updates tree.
+#[derive(Debug, Clone)]
+pub struct IncrementalMerkleSync {
```
Comment (ltchang2019) [inline nomad-base/src/merkle_sync.rs:14]: Will change to `LightMerkleSync` and use "light" in favor of "incremental" after #35 merged

File: nomad-base/src/merkle_sync.rs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+pub enum IncrementalMerkleSyncError {
+    /// IncrementalMerkleSync receives ChainCommunicationError from chain API
+    #[error(transparent)]
+    ChainCommunicationError(#[from] ChainCommunicationError),
+    /// DB Error
+    #[error("{0}")]
+    DbError(#[from] DbError),
+}
+
+impl IncrementalMerkleSync {
```
Comment (prestwich) [inline nomad-base/src/merkle_sync.rs:73]: much of this code is duplicated from `prover_sync` yes? after #35 can we replace both of these with a `TreeSync<T: Merkle<Proof = NomadProof>>`

File: nomad-base/src/merkle_sync.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            loop {
+                let tree_size = merkle.read().await.count();
+
+                info!("Waiting for leaf at index {}...", tree_size);
+                let message = db.wait_for_message(tree_size as u32).await?;
+
+                info!(
+                    index = tree_size,
+                    leaf = ?message.leaf(),
+                    "Ingesting leaf at index {}. Leaf: {}.",
```
Comment (prestwich) [inline nomad-base/src/merkle_sync.rs:140]: we're phasing out formatting in these messages. Remove the `{}` in favor of just setting the pre-message params

**Comment by ltchang2019** [comment]
Closing as other priorities/projects will make this obsolete. Ethers added RW provider, moving towards newmad in future, etc

**Pull request closed without merging**